pkg/execution/executor/executor.go

```diff
@@ -1717,17 +1720,42 @@ func (e *executor) Resume(ctx context.Context, pause state.Pause, r execution.Re
 						attribute.Bool(consts.OtelSysStepInvokeExpired, r.EventID == nil),
 					),
 				)
+				span.Send()
+				defer span.End()
+				span.SetAttributes(commonAttrs...)
 				if r.HasError() {
 					span.SetStatus(codes.Error, r.Error())
 				}
-				span.Send()
 			}
 		}
 
 		for _, e := range e.lifecycles {
 			go e.OnInvokeFunctionResumed(context.WithoutCancel(ctx), pause.Identifier, r, pause.GroupID)
 		}
 	} else {
+		if pause.StepSpanID != nil && *pause.StepSpanID != "" {
+			if spanID, err := trace.SpanIDFromHex(*pause.StepSpanID); err == nil {
+				var span *telemetry.Span
+				ctx, span = telemetry.NewSpan(ctx,
+					telemetry.WithScope(consts.OtelScopeStep),
+					telemetry.WithName(consts.OtelSpanInvoke),
+					telemetry.WithTimestamp(ts),
+					telemetry.WithSpanID(spanID),
+					telemetry.WithSpanAttributes(
+						attribute.String(consts.OtelSysStepOpcode, enums.OpcodeWaitForEvent.String()),
+						attribute.Int64(consts.OtelSysStepWaitExpires, pause.Expires.Time().UnixMilli()),
+						attribute.Bool(consts.OtelSysStepWaitExpired, r.EventID == nil),
+					),
```

Comment:
We'll also need to add some more info for the [resolvers](https://github.com/inngest/monorepo/blob/552fbeceaadf2f065d80215d59b58fc9984097df/api/schema.graphql#L1028-L1034), namely:

- Event name
- Expression
- ID of the found event (if any)